httpbp/middlewares_test.go

```diff
@@ -323,3 +324,94 @@ func TestSupportedMethods(t *testing.T) {
 		)
 	}
 }
+
+func TestInjectDebugHeaders(t *testing.T) {
```

Comment:
(sorry I didn't make it all the way through this, gotta get back to our team summit, will try to get further next round)